liota/dccs/iotcc.py

```diff
@@ -232,14 +231,22 @@ def _registration(self, msg_id, res_id, res_name, res_kind):
             }
         }
 
-    def _relationship(self, msg_id, parent_res_uuid, child_res_uuid):
+    def _relationship(self, msg_id, parent_entity, child_entity):
         return {
             "transactionID": msg_id,
             "version": self._version,
             "type": "create_relationship_request",
             "body": {
-                "parent": parent_res_uuid,
-                "child": child_res_uuid
+                "parent":  {
+                    "kind": parent_entity.entity_type,
+                    "id": parent_entity.entity_id,
+                    "name": parent_entity.name
+                },
+                "child":  {
+                    "kind": child_entity.entity_type,
+                    "id": child_entity.entity_id,
+                    "name": child_entity.name
+                }
```

Comment:
The other methods (_registration, _ properties) should be refactored to use only the entity object rather than passing all the parameters separately (res_id, res_name, res_kind).

Reply:
Yes, very good point. Avoiding last minute refactoring of existing code. We should definitely take this up in near future.